modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridDhtTopologyFutureAdapter.java

```diff
@@ -88,6 +88,10 @@ protected final CacheGroupValidation validateCacheGroup(CacheGroupContext grp, C
             return new CacheInvalidStateException(
                 "Failed to perform cache operation (cluster is not activated): " + cctx.name());
 
+        if (cctx.cache() == null)
+            return new CacheInvalidStateException(
```

Comment:
Also, does it mean that we do not distinguish cache destroy and cache restart? And would throw stop exception in both situations?

Reply:
is it important to know here cache stopped or restarting state ? or your message is: for restarting cache it would be proper to wait for restarting in this point ?

Reply:
Yes, a user should have the possibility to wait for restart finish.